presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -1520,6 +1465,86 @@ private static void testSimpleComparison(Expression expression, Symbol symbol, D
     {
         ExtractionResult result = fromPredicate(expression);
         assertEquals(result.getRemainingExpression(), TRUE_LITERAL);
-        assertEquals(result.getTupleDomain(), withColumnDomains(ImmutableMap.of(symbol, domain)));
+        TupleDomain<Symbol> actual = result.getTupleDomain();
+        TupleDomain<Symbol> expected = withColumnDomains(ImmutableMap.of(symbol, domain));
+        if (!actual.equals(expected)) {
+            fail(format("for comparison [%s] expected %s but found %s", expression.toString(), expected.toString(SESSION), actual.toString(SESSION)));
+        }
+    }
+
+    private static class NumericValues
```

Comment:
Maybe add `T` generics instead of using `Object` everywhere. This will enforce type correctness